### DISCUSSION TITLE
Separating house from associated businesses in search results

### DISCUSSION THREAD

**Discussion by rpahut** [osm-search/Nominatim#3602] [Geocoding around the world]
Searching for a house, I am having hard time separating it from the businesses that are found with it.

[On the website](https://nominatim.openstreetmap.org/ui/search.html?q=%D0%9C%D0%BE%D1%81%D0%BA%D0%B2%D0%B0+%2C++%D0%BF%D1%80%D0%BE%D1%81%D0%BF%D0%B5%D0%BA%D1%82+%D0%97%D0%B0%D1%89%D0%B8%D1%82%D0%BD%D0%B8%D0%BA%D0%BE%D0%B2+%D0%9C%D0%BE%D1%81%D0%BA%D0%B2%D1%8B+%2C++1), the addresses for the businesses found do have a specific Type, like "shop:car_repair". But same addresses, when I get them from the local Nominatim, just have the category of "place" "house". Am I required to do something to see the actual specific types?

It is also strange that the shops are returned at all, given that their addresses appear to be distinct, having an extra part on top of the house_number that is not in the query. After all, when I am searching for a city I don't get returned all the streets in it. Does Nominatim handle shops in a special way or am I doing something wrong?

**Comment by mtmail** [comment]
Some users only want postal addresses, some only the shop names. By default Nominatim returns a mix of both. On the website you link, click 'Advanced options' and play with the 'layer' parameter.

https://nominatim.org/release-docs/latest/api/Search/#result-restriction

**Reply by rpahut** [reply]
Setting layer to "address" does not filter out the shops, counterintuitively. But I have the local database imported in "address" mode, so there would be no POIs in the local results. In fact, the hairdresser and the convenience shop do not show up, only the car shop does.

I guess this might be the issue with these specific car shops. And someone has already removed them from the map.

**Comment by rpahut** [comment]
Though it still strange that the places don't have the same type in my local DB.

**Comment by lonvia** [comment]
OSM doesn't have distinct objects for addresses. They are always an attribute of another object (sometimes otherwise untagged, though). Nominatim follows the same schema because it still holds a 1:1 relationship with OSM objects.

So, when you search with layer "address" you get all objects that have a full address. In your example that amounts to one building and 4 POIs. Nominatim will not do any further filtering because it cannot really know which of the four OSM object you are really interested in.

If you have imported your local database with the "address" style, then it's not surprising you don't get the POI classifications. You haven't imported them. This style will only get the address information and classify everything as `place=house`.

**Reply by rpahut** [reply]
The addresses in my use case are of specific buildings, which is why I expect the search to match exactly one house. There shouldn't be three or five houses, and at this particular address there isn't. Some shops are being returned as houses of their own, though I would expect them to be absent from the DB along with the classifications.

No, I think there was something actually wrong with these particular car shops. Perhaps they were handled as individual places because they are located well outside of the building outline, or something like that, but matched by association as POIs. I sense there is an issue here requiring a fix. Could reimport a complete database and filter the places by category as a workaround.

**Comment by rpahut** [comment]
I will try to filter out point results that have the osm_type of 'N'. Hopefully in my area of interest all buildings will have some sort of shape to them. For a more general solution I have opened an issue #3603 .
